tests/main.rs

```diff
@@ -170,64 +252,213 @@ where
     }
 }
 
+fn mk_client_cert() -> Result<(X509, PKey<Private>), ErrorStack> {
```

Comment:
Ooof — is there really not a helper for this somewhere so we don't need all that code in here?

Reply:
I didn't find a helper for this unfortunately, so the next best thing was to remove unused bits and try to tidy & minimise this, which I have now done.